app/javascript/dashboard/routes/dashboard/conversation/Macros/MacroPreview.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div class="macro-preview">
+    <p class="macro-title">{{ macro.name }}</p>
+    <div v-for="(action, i) in resolvedMacro" :key="i" class="macro-block">
+      <div v-if="i !== macro.actions.length - 1" class="macro-block-border" />
+      <div class="macro-block-dot" />
+      <p class="macro-action-name">{{ action.actionName }}</p>
+      <p class="macro-action-params">{{ action.actionValue }}</p>
+    </div>
+  </div>
+</template>
+
+<script>
+import { AUTOMATION_ACTION_TYPES } from 'dashboard/routes/dashboard/settings/automation/constants';
+
+export default {
+  props: {
+    macro: {
+      type: Object,
+      required: true,
+    },
+  },
+  computed: {
+    resolvedMacro() {
+      return this.macro.actions.map(action => {
+        return {
+          actionName: this.getActionName(action.action_name),
+          actionValue: this.getActionValue(
+            action.action_name,
+            action.action_params
+          ),
+        };
+      });
+    },
+  },
+  methods: {
+    getActionName(key) {
+      return AUTOMATION_ACTION_TYPES.find(i => i.key === key).label;
+    },
+    resolveTeamIds(ids) {
+      const allTeams = this.$store.getters['teams/getTeams'];
+      return ids
+        .map(id => {
+          const team = allTeams.find(i => i.id === id);
+          return team ? team.name : '';
+        })
+        .join(', ');
+    },
+    resolveLabels(ids) {
+      const allLabels = this.$store.getters['labels/getLabels'];
+      return ids
```

Comment:
These items are repeated in other places in the codebase. Can we create a helper or something to avoid duplicating this in different places?